Automattic/wp-calypso
# Checkout: Decouple existingCardProcessor from checkout

#### Changes proposed in this Pull Request

This moves the composite checkout payment processor function `existingCardProcessor` to its own file and converts it to TypeScript. This is the first payment processor to be converted to TypeScript so it will set the stage for the rest to be converted as well by cleaning up the types and API. Notably it creates a `PaymentProcessorOptions` type which is the data passed into each processor.

This also refactors the processor a bit to decouple it from the 'wpcom' `@wordpress/data` Store so it can be used in other parts of calypso more easily. It does this mostly by fetching data it needs within the closure inside `CompositeCheckout`, making it possible for `existingCardProcessor` to rely only on its inputs.

The reason we are starting with this processor is that it can be used for "one-click checkout" (see #48132).

#### Testing instructions

- Make sure your browser devtools are recording network calls, and make sure that those calls are preserved after a redirect.
- Add a plan with a domain to your cart and visit checkout.
- Complete checkout using a saved credit card.
- Verify that the transaction completes as expected and that you are redirected as expected.
- Verify that the network request to the `/me/transactions` endpoint includes `domainDetails`.
